serverless/experimental/wasm/main.go

```diff
@@ -163,32 +187,95 @@ func queueLeaf() js.Func {
 	return jsonFunc
 }
 
-func monitor() {
+func monitor(ctx context.Context, f client.Fetcher) {
 	cpCur := &logfmt.Checkpoint{}
-
+	logProofVerifier := logverifier.New(rfc6962.DefaultHasher)
 	for {
-		<-time.Tick(time.Second)
-		cpRaw, _ := webstorage.ReadCheckpoint(logPrefix)
-		cp := &logfmt.Checkpoint{}
-		if _, err := cp.Unmarshal(cpRaw); err != nil {
-			monMsg(fmt.Sprintf("Couldn't parse CP: %v", err))
+		select {
+		case <-time.Tick(time.Second):
+		case <-ctx.Done():
+			return
+		}
+
+		cp, _, err := client.FetchCheckpoint(ctx, f, logVer)
+		if err != nil {
+			if !errors.Is(err, os.ErrNotExist) {
+				monMsg(err)
+			}
+			// CP likely just doesn't exist yet - no sequence/integrate run has happened.
 			continue
 		}
 
-		if cp.Size > cpCur.Size {
-			monMsg("----------------------------------")
-			monMsg(fmt.Sprintf("Saw new CP:\n%s", string(cp.Marshal())))
-			cpCur = cp
+		if cp.Size <= cpCur.Size {
+			continue
 		}
+		monMsg("----------------------------------")
+		monMsg(fmt.Sprintf("<invert>Saw new CP with size %d</invert>", cp.Size))
 
+		if cpCur.Size > 0 {
+
+			pb, err := client.NewProofBuilder(ctx, *cp, rfc6962.DefaultHasher.HashChildren, f)
+			if err != nil {
+				monMsg(err)
+				continue
+			}
+			proof, err := pb.ConsistencyProof(ctx, cpCur.Size, cp.Size)
+			if err != nil {
+				monMsg(err)
+				continue
+			}
+			if err := logProofVerifier.VerifyConsistencyProof(int64(cpCur.Size), int64(cp.Size), cpCur.Hash, cp.Hash, proof); err != nil {
+				monMsg(err)
+				continue
+			}
+			monMsg("Proof:")
+			monMsg(logfmt.Proof(proof).Marshal())
+			monMsg("New CP verified to be consistent")
+		}
+		cpCur = cp
+	}
+}
+
+func initKeys() {
+	var s, v string
+	var err error
+
+	prevV := js.Global().Get("sessionStorage").Call("getItem", "log.pub")
+	prevS := js.Global().Get("sessionStorage").Call("getItem", "log.sec")
```

Comment:
It's blindingly obvious, but perhaps another comment here saying this is *A DEMO* and terrible idea for production would be useful.

Reply:
hehe, ok